### ISSUE TITLE
`lightkurve.SearchResult.download_all` fails for heterogeneous data products?

### ISSUE THREAD

**Issue by lgbouma** [lightkurve/lightkurve#1173]
#### Problem description
One might expect that for a given TIC ID, calling`lk.search_lightcurve(<TIC ID>).download_all()` would download all the available light curves on MAST for that TIC ID.  There are some objects for which this is not the case.

#### Example

```python
import lightkurve as lk
lk.search_lightcurve('TIC 89463560').download_all()
```

#### Expected behavior
I would expect the above to download all the available light curves for the listed star -- in this particular case, there are 19 available when I run this (2022/03/09).

What actually happens is the stack trace listed below.

One workaround that seems to work is to select light curves based on a given desired author, or desired exposure time:

```
ticstr = 'TIC 89463560'
lcset = lk.search_lightcurve(ticstr)
lcc = lcset[(lcset.author=='SPOC') & (lcset.exptime.value==120)].download_all()
```

This, along with the details of the `time_column` error listed in the stack-trace below, would suggest that it's something in the resolution of the heterogeneous column names for heterogeneous HLSP light curve sources at MAST that is the source of the error.

I'll mention that this is basically the same behavior encountered in https://github.com/lightkurve/lightkurve/issues/1123 / https://github.com/lightkurve/lightkurve/issues/1124, though perhaps in a more generic use-case.

**Many thanks** for maintaining this very useful piece of software!

#### Environment

-  platform: OSX
-  lightkurve version: 2.1.0
-  installation method: conda

#### Stack trace

```
WARNING: File may have been truncated: actual file length (131072) is smaller than the expected size (383040) [astropy.io.fits.file]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/timeseries/sampled.py in read(self, filename, time_column, time_format, time_scale, format, *args, **kwargs)
    365             # First we try the readers defined for the BinnedTimeSeries class
--> 366             return super().read(filename, format=format, *args, **kwargs)
    367

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/table/connect.py in __call__(self, *args, **kwargs)
     60
---> 61         out = registry.read(cls, *args, **kwargs)
     62

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/registry.py in read(cls, format, cache, *args, **kwargs)
    519         reader = get_reader(format, cls)
--> 520         data = reader(*args, **kwargs)
    521

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/io/tess.py in read_tess_lightcurve(filename, flux_column, quality_bitmask)
     33     """
---> 34     lc = read_generic_lightcurve(filename, flux_column=flux_column, time_format="btjd")
     35

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/io/generic.py in read_generic_lightcurve(filename, time_column, flux_column, flux_err_column, quality_column, cadenceno_column, centroid_col_column, centroid_row_column, time_format, ext)
     39         validate_method(ext, supported_methods=[hdu.name.lower() for hdu in hdulist])
---> 40     tab = Table.read(hdulist[ext], format="fits")
     41

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/table/connect.py in __call__(self, *args, **kwargs)
     60
---> 61         out = registry.read(cls, *args, **kwargs)
     62

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/registry.py in read(cls, format, cache, *args, **kwargs)
    519         reader = get_reader(format, cls)
--> 520         data = reader(*args, **kwargs)
    521

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/fits/connect.py in read_table_fits(input, hdu, astropy_native, memmap, character_as_bytes)
    242     # col.array to make sure that the data is scaled correctly if needed.
--> 243     data = table.data
    244

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/utils/decorators.py in __get__(self, obj, owner)
    757             else:
--> 758                 val = self.fget(obj)
    759                 obj.__dict__[self._key] = val

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/fits/hdu/table.py in data(self)
    398     def data(self):
--> 399         data = self._get_tbdata()
    400         data._coldefs = self.columns

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/fits/hdu/table.py in _get_tbdata(self)
    171             raw_data = self._get_raw_data(self._nrows, columns.dtype,
--> 172                                           self._data_offset)
    173             if raw_data is None:

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/fits/hdu/base.py in _get_raw_data(self, shape, code, offset)
    519         elif self._file:
--> 520             return self._file.readarray(offset=offset, dtype=code, shape=shape)
    521         else:

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/io/fits/file.py in readarray(self, size, offset, dtype, shape)
    330                 return np.ndarray(shape=shape, dtype=dtype, offset=offset,
--> 331                                   buffer=self._mmap)
    332             else:

TypeError: buffer is too small for requested array

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-1-390e7a97bbf4> in <module>
      1 import lightkurve as lk
----> 2 lk.search_lightcurve('TIC 89463560').download_all()

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/utils.py in wrapper(*args, **kwargs)
    553             sys.stdout = devnull
    554             try:
--> 555                 return f(*args, **kwargs)
    556             # restore to default
    557             finally:

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/search.py in download_all(self, quality_bitmask, download_dir, cutout_size, **kwargs)
    426                     download_dir=download_dir,
    427                     cutout_size=cutout_size,
--> 428                     **kwargs,
    429                 )
    430             )

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/search.py in _download_one(self, table, quality_bitmask, download_dir, cutout_size, **kwargs)
    288                 )["Local Path"][0]
    289                 log.debug("Finished downloading.")
--> 290             return read(path, quality_bitmask=quality_bitmask, **kwargs)
    291
    292     @suppress_stdout

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/io/read.py in read(path_or_url, **kwargs)
     90         return KeplerLightCurve.read(path_or_url, format="kepler", **kwargs)
     91     elif filetype == "TessLightCurve":
---> 92         return TessLightCurve.read(path_or_url, format="tess", **kwargs)
     93     elif filetype == "QLP":
     94         return TessLightCurve.read(path_or_url, format="qlp", **kwargs)

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/lightkurve/lightcurve.py in read(cls, *args, **kwargs)
   3172         if kwargs.get("format") is None:
   3173             kwargs["format"] = "tess"
-> 3174         return super().read(*args, **kwargs)
   3175
   3176     def to_fits(

~/local/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/timeseries/sampled.py in read(self, filename, time_column, time_format, time_scale, format, *args, **kwargs)
    371
    372             if time_column is None:
--> 373                 raise ValueError("``time_column`` should be provided since the default Table readers are being used.")
    374
    375             table = Table.read(filename, format=format, *args, **kwargs)

ValueError: ``time_column`` should be provided since the default Table readers are being used.
```

**Comment by orionlee** [comment]
Hmm, I cannot reproduce the problem with `TIC 89463560` as suggested. For me , all 21 products are downloaded successfully (as of 2022/03/28) with a mix of SPOC, QLP, and CDIPs, all of which are supported by Lightkurve.

I have 2 suggestions:
1. The warning at the top of the stacktrace suggested that downloading of some FITS file may been incomplete. I vaguely recall there were similar issues before (Lightkurve giving unhelpful errors while reading a corrupted / incomplete FIS file). 
You could try to locate the offending FITS file, or you can delete all existing cached FIS files (default location: `~/.lightkurve-cache/`). The problem should go away.

2. If removing offending FITS file does not help, please include astropy version in the report, in case it is due to some incompatibility.


Reference. I downloaded all of the products of `TIC 89463560`  successfully with
```
Lightkurve:  2.1.2dev   (basically the same as the just released 2.1.1)
Astropy:  5.0.3
OS: Windows 11
```

**Comment by lgbouma** [comment]
Hmm, interesting.  If it's resolved in the 2.1.2dev / astropy 5.0.3 release, then happy to not spend too much time digging down the exact root cause.  Thank you for looking into it, regardless!

**Comment by barentsen** [comment]
@lgbouma Thank you for opening the issue! 👍 
@orionlee Thank you for the investigation and helpful tips! 🙏 

Just to confirm, the example above also works fine for me using the latest version of Lightkurve (v2.1.1).  Screenshot:

<img width="851" alt="Screen Shot 2022-03-28 at 8 09 20 PM" src="https://user-images.githubusercontent.com/817669/160525323-07dcb853-9155-43cb-8d59-c1a6fdc60398.png">

I suspect the error was caused by a partial/corrupt download.  Removing the cache directory (`~/.lightkurve-cache/`) as suggested by @orionlee should make the error go away in that scenario.
